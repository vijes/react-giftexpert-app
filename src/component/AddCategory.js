import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories } ) => {

    const [inputvalue, setinputvalue] = useState('')

    const handleInputChange = ( e ) => {
        // console.log(e.target.value);
        setinputvalue( e.target.value );
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if ( inputvalue ) {
            setCategories( cats => [ inputvalue, ...cats ]);
            setinputvalue('');
        }else{
            console.log('No debe ser vacio el input');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <p> { inputvalue } </p>
            <input 
                type="text"
                value={ inputvalue }
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}