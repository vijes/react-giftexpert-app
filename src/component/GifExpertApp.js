import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {

    // const categories = ['One punch', 'Samurai X', 'Dragon Ball'];

    // const [categories, setCategories] = useState(['Samurai X']);
    const [categories, setCategories] = useState( defaultCategories );

    // const handleAdd = () => {
    //     setCategories( ['Dragon Ball GT', ...categories] );
    //     // setCategories( cats => [ ... cats, 'Dagon Ball GT']);
    // }

    return (
        <>
            <h2>GiftExpertApp</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr />

            <ol>
                {
                    categories.map( category => (
                        <GifGrid 
                            key={ category }
                            category={ category } 
                        />
                    ))
                }
            </ol>
        </>
    );
}

export default GifExpertApp;