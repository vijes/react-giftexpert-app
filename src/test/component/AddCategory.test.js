import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../component/AddCategory';
import '@testing-library/jest-dom'

describe('Pruebas en componente <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories } />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories } />);
    });

    test('debe mostrarse correctamente', () => {

       expect( wrapper ).toMatchSnapshot(); 
        
    });

    test('Debe cambiar la caja de texto', () => {
        
        const input = wrapper.find('input');

        const value = 'Hola mundo, test Victor';
        input.simulate('change', { target: { value } });

        expect( wrapper.find('p').text().trim() ).toBe( value );
    });

    test('no debe de postear la informacion con submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){} });
        
        expect( setCategories ).not.toHaveBeenCalled();        
    });

    test('debe de llamar el setcatgories y limpiar la caja de texto. ', () => {

        const value = 'Hola mundo, test completa';
        wrapper.find('input').simulate('change', { target: { value } });
        // console.log('Inputchange ejecutado: ', wrapper.find('p').text().trim() );

        wrapper.find('form').simulate('submit', { preventDefault(){} });

        // console.log('El setCategories debio llamarse al menos una vez');
        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );

        // console.log('El input value debe estar vacio ');
        expect( wrapper.find('input').prop('value') ).toBe( '' );
    })
    
});
