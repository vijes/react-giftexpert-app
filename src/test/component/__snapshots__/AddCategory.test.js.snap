// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <AddCategory /> debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
